test/xds/xds_client_outlier_detection_test.go

```diff
@@ -107,52 +109,91 @@ func clusterWithOutlierDetection(clusterName, edsServiceName string, secLevel e2
 		MaxEjectionPercent:             &wrapperspb.UInt32Value{Value: 1},
 		FailurePercentageThreshold:     &wrapperspb.UInt32Value{Value: 50},
 		EnforcingFailurePercentage:     &wrapperspb.UInt32Value{Value: 100},
-		FailurePercentageRequestVolume: &wrapperspb.UInt32Value{Value: 1},
-		FailurePercentageMinimumHosts:  &wrapperspb.UInt32Value{Value: 1},
+		FailurePercentageRequestVolume: &wrapperspb.UInt32Value{Value: 8},
+		FailurePercentageMinimumHosts:  &wrapperspb.UInt32Value{Value: 3},
 	}
 	return cluster
 }
 
+// checkRoundRobinRPCs verifies that EmptyCall RPCs on the given ClientConn,
+// connected to a server exposing the test.grpc_testing.TestService, are
+// roundrobined across the given backend addresses.
+//
+// Returns a non-nil error if context deadline expires before RPCs start to get
+// roundrobined across the given backends.
+func checkRoundRobinRPCs(ctx context.Context, client testpb.TestServiceClient, addrs []resolver.Address) error {
```

Comment:
Is it not possible to use this helper function here: https://github.com/grpc/grpc-go/blob/0fe49e823fcd9904afba6cd5e5980da4390d1899/internal/testutils/roundrobin/roundrobin.go#L83

Reply:
Unfortunately, this is slightly different logic than that helper. This line https://github.com/grpc/grpc-go/blob/0fe49e823fcd9904afba6cd5e5980da4390d1899/internal/testutils/roundrobin/roundrobin.go#L59 expects that an empty RPC made to one of the backends doesn't return an error, but for the sake of this test, a backend needs to return an error for the testing functionality. This, however, is the same exact helper as https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/outlierdetection/e2e_test/outlierdetection_test.go#L102. I don't like the duplicated code, but IIRC, I put it there to avoid a cyclic dependency. Do you have any suggestions where to put this shared helper?